### PR TITLE
JS: Move `js/log-injection` into non-experimental. 

### PR DESCRIPTION
**Pull request by erik-krogh**
I added API-graphs to `Logging.qll` such that the manual property-tracking implemented in `LogInjection.qll` was no longer needed.

### CODE REVIEW

**Comment by asgerf** [comment]
Could you run an evaluation? I believe this is the first use of `API::EntryPoint` to be merged into main so it's worth checking.

**Comment by erik-krogh** [comment]
[An evaluation on `nightly` for a few default queries](https://git.semmle.com/erik/dist-compare-reports/tree/eriks-profile-machine_1607320762891) looks fine. 

And [profilling just `LogInjection.ql`](https://git.semmle.com/erik/dist-compare-reports/tree/eriks-profile-machine_1607349029433) shows that performance is not completely terrible for the `LogInjection` query itself.

**Comment by erik-krogh** [comment]
@asgerf Can I get an approve?

**Review by asgerf — APPROVED** [review]

**Comment by erik-krogh** [comment]
@mchammer01 can I get a doc review? 
Mostly of the [qhelp file](https://github.com/github/codeql/blob/c50951cbae457aedd92e75e2bdd31ca2d333becc/javascript/ql/src/Security/CWE-117/LogInjection.qhelp) that was moved out of experimental.

**Comment by mchammer01** [comment]
@erik-krogh - will try to look at this later today (I have urgent things to do for the GHES 30 docs ship first).

File: javascript/ql/src/Security/CWE-117/LogInjection.ql [hunk 1]
```diff
@@ -4,15 +4,15 @@
  *              insertion of forged log entries by a malicious user.
  * @kind path-problem
  * @problem.severity error
- * @precision high
+ * @precision medium
```
Comment (mchammer01) [inline javascript/ql/src/Security/CWE-117/LogInjection.ql:7]: On line 2, please use sentence case as per [this link](https://github.com/github/codeql/blob/8d4f7e2db7ec3b7559a3e358fd1b8a7203ce7474/docs/query-metadata-style-guide.md#query-name-name).

Comment (mchammer01) [reply javascript/ql/src/Security/CWE-117/LogInjection.ql:7]: Oops please ignore this comment for now, I've just started a review (or at least, that's what I tried to do!).

**Review by mchammer01 — COMMENTED** [review]
@erik-krogh - this LGTM 🙂 , I have a few comments on the qhelp file but because the file has been moved without any changes, I couldn't comment on the diff, so please find my comments below. As usual, feel free to ignore if you think some aren't relevant.

1. "If unsanitized user input is written to a log entry, a malicious user may be able to forge new log entries." seems a bit like a repetition of the query name that appears just above. I'd perhaps suggest removing it?

2. "If the log entries are plain text then line breaks should be removed from user input, using String.prototype.replace or similar. Care should also be taken that user input is clearly marked in log entries" - remove the "and that a malicious user cannot cause confusion in other ways.", I think it's implied, and it doesn't add anything new.

3. "If the log entries are plain text " -> "If the log entries are written in plain text" or "If the log entries are in plain text"

4. "For log entries that will be displayed in HTML, user input should be HTML-encoded before being logged, to prevent forgery and other forms of HTML injection." - add an hyphen between `HTML` and `encoded`.

5. "In the second case, the username is used to build an error", add a comma between "case" and "the username".

Hope this helps!

**Comment by erik-krogh** [comment]
Thanks for the review.   
I've applied all the suggested changes except for `1.`. 

Can I get a final approve @mchammer01 and @asgerf?

**Review by asgerf — APPROVED** [review]

**Review by mchammer01 — APPROVED** [review]
Thank you for the updates  @erik-krogh ✨ 
LGTM from my point of view 👍🏻

**Pull request merged**